src/errors/BundlrError.ts

```diff
@@ -0,0 +1,51 @@
+import { MetaplexError, MetaplexErrorInputWithoutSource } from './MetaplexError';
+
+export class BundlrError extends MetaplexError {
+  constructor(input: MetaplexErrorInputWithoutSource) {
+    super({
+      ...input,
+      key: `plugin.bundlr.${input.key}`,
+      title: `Bundlr > ${input.title}`,
+      source: 'plugin',
+      sourceDetails: 'Bundlr',
+    });
+  }
+
+  static failedToInitializeBundlr(cause: Error) {
+    return new this({
```

Comment:
In general I'm not a huge fan of `this` business. Afaik most linters will complain as well.
To me this would read much clearer if this was `new BundlrError(...`

However I would not even expose errors as static methods since now you cannot easily `instanceof` check them (you worked around this with a `key` property, however the language already provides a mechanism to detect custom error types).

As an example the above I'd implement as follows:

```ts
class FailedToInitializeBundlrError extends BundlrError {
  constructor(cause: Error) {
    super({
      cause,
      title: 'Failed to Initialize Bundlr',
      problem: 'Bundlr could not be initialized.',
      solution:
        'This could happen for a variety of reasons. ' +
        'Check the underlying error below for more information.',
    });
  }
}
```

Which then allows you to do something similar to:

```ts
if (err instanceof FailedToInitializeBundlrError) {
  console.error(err.cause);
}
```

Now you get a separate type for each error which is much closer to how custom errors in JS are supposed to be implemented (see [mdn custom error section](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error#custom_error_types)).

Also in a case that this is _thrown_ it will show up properly as `FailedToInitializeBundlrError` in the output.

Reply:
Good point! I'll make the changes